source_gen/lib/src/builder.dart

```diff
@@ -281,36 +280,31 @@ class LibraryBuilder extends _Builder {
         );
 }
 
-Stream<GeneratedOutput> _generate(
+Stream<String> _generate(
   LibraryElement library,
   List<Generator> generators,
   BuildStep buildStep,
 ) async* {
   final libraryReader = LibraryReader(library);
   for (var i = 0; i < generators.length; i++) {
     final gen = generators[i];
-    try {
-      var msg = 'Running $gen';
-      if (generators.length > 1) {
-        msg = '$msg - ${i + 1} of ${generators.length}';
-      }
-      log.fine(msg);
-      var createdUnit = await gen.generate(libraryReader, buildStep);
-
-      if (createdUnit == null) {
-        continue;
-      }
+    var msg = 'Running $gen';
+    if (generators.length > 1) {
+      msg = '$msg - ${i + 1} of ${generators.length}';
+    }
+    log.fine(msg);
+    var createdUnit = await gen.generate(libraryReader, buildStep);
 
-      createdUnit = createdUnit.trim();
-      if (createdUnit.isEmpty) {
-        continue;
-      }
+    if (createdUnit == null) {
+      continue;
+    }
 
-      yield GeneratedOutput(gen, createdUnit);
-    } catch (e, stack) {
```

Comment:
I mentioned this on an internal thread - if we want to keep the behavior of emitting logs for all errors and not just the first one we could keep the try/catch and the `log.severe`.

I don't have strong opinions on whether we'd prefer multiple errors or to fail fast on the first one. @jakemac53 might have an opinion.

For now I'd lean towards this implementation and fail fast.

Reply:
The issue with logging is that it will potentially swallow build failures intended for the build runner to catch - I'm not sure how we can get both properties.

Reply:
Logging at severe _does_ cause a build failure. The difference are:

 - throw:
   - fail fast
   - error output is based on `toString()` for the exception and not controlled by builder author.
   - May be caught somewhere else in builder author code and ignored (won't be caught and ignored by build system or `source_gen`).
- `log.severe`:
   - fail slow - can emit multiple error messages, but the overall build step will still fail.
   - error output is under your control, not necessary an exception `toString()`.
   - can't be caught and ignored. Every step that logs at severe will be a failure.

Reply:
catch all is an anti-pattern I'm happy to ditch. Also guessing that the throw case is easier to debug.

Reply:
I'll just hold off landing this until @jakemac53 can weigh in

Reply:
Oh right, so whether we log a build failure here or further down in the build runner doesn't actually matter. I was thinking logging it here didn't give the underlying build system a chance to handle it, but as long as that handling takes places on the log record, it's fine.